### PR TITLE
refactor(spacing): generate spacing atomic classes programatically

### LINKED ISSUE #1653: Responsive class `.pb0` does not work
Noticed `sm:pb0` etc does not work. Looks like an oversight in setting up the classes

https://github.com/StackExchange/Stacks/blob/64f0bf227e4bbcc3edf50e11fdbfce87df81f569/lib/atomic/spacing.less#L210-L224

### PR DESCRIPTION
**Pull request by dancormier**
This PR expands on https://github.com/StackExchange/Stacks/pull/1652 and closes https://github.com/StackExchange/Stacks/issues/1653

---

Spacing class generation is prone to bugs and accidental omissions. This PR make generation of spacing (and incidentally, positioning) atomic classes and documentation more programmatic.

### CODE REVIEW

File: lib/atomic/__snapshots__/spacing.less.test.ts.snap
Comment (giamir) [inline lib/atomic/__snapshots__/spacing.less.test.ts.snap:1]: @dancormier just glancing at this PR. What would you think about adding the less snapshot test directly in develop to start with to then have here a better diff of what have actually changed with your refactor?

**Comment by giamir** [comment]
This refactor makes a lot of sense to me. I just glanced at the PR and left a comment. I will do a full review once you mark the PR as ready for review. Thanks @dancormier

File: lib/atomic/__snapshots__/spacing.less.test.ts.snap
Comment (dancormier) [reply lib/atomic/__snapshots__/spacing.less.test.ts.snap:1]: Done in this PR https://github.com/StackExchange/Stacks/pull/1657

Comment (giamir) [inline lib/atomic/__snapshots__/spacing.less.test.ts.snap:1]: @dancormier Not perfect but at least we have a diff of what changed now. Let me know when this PR is ready for a final review.

Comment (dancormier) [reply lib/atomic/__snapshots__/spacing.less.test.ts.snap:1]: It's ready for your review now but no rush on this one

File: docs/_data/spacing.json [hunk 1]
```diff
@@ -1,446 +1,109 @@
...
+                "define": "margin x-axis (left, right)",
+                "responsive": true
+            },
+            {
+                "term": "my",
+                "define": "margin y-axis (top, bottom)",
+                "responsive": true
+            },
+            {
+                "term": "mxn",
```
Comment (giamir) [inline docs/_data/spacing.json:42]: Looks like `myn` is missing from this table. It was an issue also before these changes but we should correc it as part of this PR.
Screenshot 2024-03-01 at 14 46 19

**Review by giamir — APPROVED** [review]
I took the liberty to push [this commit](https://github.com/StackExchange/Stacks/pull/1654/commits/01b59cfadc924b3c2f3b2d5872cbaba7e7b63b86) and made a small observation about the docs missing `myn`.

Thanks for doing this refactor. I am good to merge when you are. 🙏

**Pull request merged**